packages/record/src/recorder.ts

```diff
@@ -1,58 +1,106 @@
 import Data from "@lookit/data";
-import lookitS3 from "@lookit/data/dist/lookitS3";
+import {
+  default as lookitS3,
+  default as LookitS3,
+} from "@lookit/data/dist/lookitS3";
 import autoBind from "auto-bind";
 import { JsPsych } from "jspsych";
 import Mustache from "mustache";
+import play_icon from "../img/play-icon.svg";
+import record_icon from "../img/record-icon.svg";
+import playbackFeed from "../templates/playback-feed.mustache";
 import webcamFeed from "../templates/webcam-feed.mustache";
 import {
+  CreateURLError,
   MicCheckError,
+  NoPlayBackElementError,
   NoStopPromiseError,
   NoStreamError,
+  NoWebCamElementError,
   RecorderInitializeError,
-} from "./error";
+  S3UndefinedError,
+  StreamActiveOnResetError,
+  StreamDataInitializeError,
+  StreamInactiveInitializeError,
+} from "./errors";
 import { CSSWidthHeight } from "./types";
 // import MicCheckProcessor from './mic_check';  // TO DO: fix or remove this. See: https://github.com/lookit/lookit-jspsych/issues/44
 
 /** Recorder handles the state of recording and data storage. */
 export default class Recorder {
+  private url?: string;
+  private _s3?: lookitS3;
+
   private blobs: Blob[] = [];
   private localDownload: boolean =
     process.env.LOCAL_DOWNLOAD?.toLowerCase() === "true";
-  private filename: string;
-  private stopPromise: Promise<void> | undefined;
+  private filename?: string;
   private minVolume: number = 0.1;
-  private webcam_element_id = "lookit-jspsych-webcam";
   public micChecked: boolean = false;
+
   /**
    * Use null rather than undefined so that we can set these back to null when
    * destroying.
    */
   private processorNode: AudioWorkletNode | null = null;
-  private s3: lookitS3 | null = null;
-  /**
-   * Store the reject function for the stop promise so that we can reject it in
-   * the destroy recorder method.
-   */
-  private rejectStopPromise: (reason: string) => void = () => {};
+
+  private stopPromise?: Promise<void>;
+  private webcam_element_id = "lookit-jspsych-webcam";
+  private playback_element_id = "lookit-jspsych-playback";
+
+  private streamClone: MediaStream;
 
   /**
    * Recorder for online experiments.
    *
    * @param jsPsych - Object supplied by jsPsych.
-   * @param fileNamePrefix - Prefix for the video recording file name (string).
-   *   This is the string that comes before "_<TIMESTAMP>.webm".
    */
-  public constructor(
-    private jsPsych: JsPsych,
-    fileNamePrefix: string,
-  ) {
-    this.filename = this.createFilename(fileNamePrefix);
-    if (!this.localDownload) {
-      this.s3 = new Data.LookitS3(this.filename);
-    }
+  public constructor(private jsPsych: JsPsych) {
+    this.streamClone = this.stream.clone();
```

Comment:
I'm confused... How can `this.stream` exist when the Recorder is first instantiated? 
We're now using the Recorder to initialize the jsPsych recorder (instead of the jsPsych `initialize-camera` and `initialize-microphone` plugins) so we need to be able to create a Recorder instance before the stream is initialized.

Reply:
Following up on our conversation about this: it will be a problem for my current version of the `video-config` plugin, but I can fix it by moving some of the methods out of the `Recorder` class and into the video config class - that way I won't need to instantiate a Recorder until the stream has been initialized with jsPsych. 

If that turns out to be too hard for some reason, then I'll put in a conditional here to make sure the stream exists before it is assigned to `streamClone`. 

Marking this as resolved since I can fix it one way or another 👍